pgmax/fg/graph.py

```diff
@@ -180,31 +183,66 @@ def _register_factor_group(
                 self._factor_group_to_msgs_starts[factor_group]
                 + factor_num_states_cumsum[vv]
             )
-            self._factor_to_potentials_starts[factor] = (
-                self._factor_group_to_potentials_starts[factor_group]
-                + vv * factor.log_potentials.shape[0]
-            )
-            factor_group_num_configs += factor.log_potentials.shape[0]
+
+            if not isinstance(factor_group, groups.ORFactorGroup):
+                self._factor_to_potentials_starts[factor] = (
+                    self._factor_group_to_potentials_starts[factor_group]
+                    + vv * factor.log_potentials.shape[0]
+                )
+                factor_group_num_configs += factor.log_potentials.shape[0]
 
         self._total_factor_num_states += factor_num_states_cumsum[-1]
-        self._total_factor_num_configs += factor_group_num_configs
         if name is not None:
             self._named_factor_groups[name] = factor_group
 
+        if not isinstance(factor_group, groups.ORFactorGroup):
+            self._total_factor_num_configs += factor_group_num_configs
+
     @cached_property
-    def wiring(self) -> nodes.EnumerationWiring:
+    def enum_wiring(self) -> None or nodes.EnumerationWiring:
         """Function to compile wiring for belief propagation.
 
         If wiring has already beeen compiled, do nothing.
 
         Returns:
             Compiled wiring from individual factors.
+            None if the graph only has OR factors.
         """
-        wirings = [
-            factor.compile_wiring(self._vars_to_starts) for factor in self.factors
+        if np.all([isinstance(factor, nodes.ORFactor) for factor in self.factors]):
+            return None
+
+        enum_wirings = [
+            factor.compile_wiring(self._vars_to_starts)
+            for factor in self.factors
+            if isinstance(factor, nodes.EnumerationFactor)
         ]
-        wiring = fg_utils.concatenate_enumeration_wirings(wirings)
-        return wiring
+        enum_wiring = fg_utils.concatenate_enumeration_wirings(enum_wirings)
```

Comment:
You need to be careful about this concatenation here. Since there might be both enumeration factors and OR factors present, you might no longer be able to rely on simple offsets to do this concatenation.